### PR TITLE
Wheel PCB - MPU Integration

### PR DESCRIPTION
**Pull request by nwdepatie**
Made some changes to change the drive state of the car based on the wheel IO inputs, also added the ability to decode those inputs and debounce the signals. Might wanna test on the car before we merge in

### CODE REVIEW

File: MPU/src/driverio.cpp [hunk 1]
```diff
@@ -3,99 +3,119 @@
...
+    {
+        driveState = (driveState - 1) % MAX_STATES;
+        Serial.println("Decrement!");
+        changeStateTimer.startTimer(CHANGE_STATE_TIME);
+    }
+
     //If the motor controller is on OR the motor is off and it is ready to go
-    if(motorController->getIsOn() || 
-        (!motorController->getIsOn() && !motorController->checkFault()))
+    if(canTransitionToOn() || canTransitionToOff())
```
Comment (mccauleyma) [inline MPU/src/driverio.cpp:62]: This does not work. Speaker just stays on constantly lol

File: MPU/src/pedals.cpp [hunk 2]
```diff
@@ -87,7 +87,7 @@ int16_t Pedals::calcTorque(double torqueScale)
 
 	pedalTorque = torqueScale * MAXIMUM_TORQUE;
 
-	int16_t torqueLim = calcCLTorqueLimit();
+	int16_t torqueLim = 100;//calcCLTorqueLimit();
```
Comment (mccauleyma) [inline MPU/src/pedals.cpp:90]: 🤨

File: MPU/src/mpu.cpp [hunk 3]
```diff
@@ -65,7 +63,7 @@ bool isCANLineOK()
 
 void setCANLineOK()
 {
-    canTest_wait.startTimer(1500);
+    canTest_wait.startTimer(2000);
```
Comment (mccauleyma) [inline MPU/src/mpu.cpp:66]: Why?

File: MPU/src/main.cpp [hunk 4]
```diff
@@ -3,13 +3,18 @@
...
+#include "mpuConfig.h"
+
+int driveState;
 
 void setup()
 {
     NERduino.begin();
-    initializeCAN(1);           //The "1" parameter is useless for now, in the future the parameter is which CAN line to initialize
+    driveState = OFF;
+    initializeCAN(CANLINE_1, BAUD_RATE, &(*mpuCanCallback));
```
Comment (mccauleyma) [inline MPU/src/main.cpp:17]: Is `&(*mpuCanCallback)` always how we pass the callback function?

**Review by mccauleyma — CHANGES_REQUESTED** [review]
Generally seems ok, however some of the logic is def off. Currently the car does not turn on, it just makes the RTDS constantly (ouch). Also I am realizing, do we reset the states when TSMS flips? Cause we need to reset to the off state when that happens and Idk if that is happening.

File: MPU/src/main.cpp (hunk 4 above)
Comment (nwdepatie) [reply MPU/src/main.cpp]: Yeah this is the new way of doing it with the updated CAN framework, it works well and is tested

File: MPU/src/mpu.cpp (hunk 3 above)
Comment (nwdepatie) [reply MPU/src/mpu.cpp]: BMS integration, we can decrease it if it works

File: MPU/src/pedals.cpp (hunk 2 above)
Comment (nwdepatie) [reply MPU/src/pedals.cpp]: I can fix

**Comment by nwdepatie** [comment]
Just made some new changes, but I need to validate hardware

**Review by mccauleyma — APPROVED** [review]
Worked in testing. Will need iteration, but lets get this merged first

**Pull request merged**